cmake/mission_build.cmake

```diff
@@ -0,0 +1,487 @@
+  endforeach(DEP ${MISSION_DEPS})
+
+  # Doxygen-based documentation generation targets
+  # Create a directory for documentation output
+  file(MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/doc")
+
+  # Generate a customized Doxyfile file for the Doxygen docs.
+  # This file must be present in the directory where "doxygen" is executed
+  # If the user has provided a "Doxyfile" in their top level documentation directory,
+  # then assume they have also set PROJECT_NAME and PROJECT_BRIEF in that.
+  # Otherwise, generate reasonable strings for these values.
+  set(MISSION_DOXYFILE_USER_CONTENT)
+  if (EXISTS "${MISSION_SOURCE_DIR}/doc/Doxyfile")
+    list(APPEND MISSION_DOXYFILE_USER_CONTENT "@INCLUDE = ${MISSION_SOURCE_DIR}/doc/Doxyfile\n")
+  endif (EXISTS "${MISSION_SOURCE_DIR}/doc/Doxyfile")
+
+  foreach(APP ${MISSION_DEPS})
+    # OSAL is handled specially, as only part of it is used
+    if (NOT APP STREQUAL "osal")
+      if (EXISTS "${${APP}_MISSION_DIR}/docs/${APP}.doxyfile.in")
+        # If the module provides its own doxyfile, then include it directly
+        # This allows for app-specific fine-tuning of the sources, based on its own source tree
+        configure_file("${${APP}_MISSION_DIR}/docs/${APP}.doxyfile.in"
+            "${CMAKE_BINARY_DIR}/doc/${APP}.doxyfile")
+        list(APPEND MISSION_DOXYFILE_USER_CONTENT "@INCLUDE = ${CMAKE_BINARY_DIR}/doc/${APP}.doxyfile\n")
+      else()
+        # Otherwise just add this entire directory to the "INPUT" list
+        list(APPEND MISSION_DOXYFILE_USER_CONTENT "INPUT += ${${APP}_MISSION_DIR}\n")
+      endif()
+    endif()
+  endforeach()
+
+  # In all cases it is assumed to include the CFE documentation as well (could be configurable?)
+  file(WRITE "${CMAKE_BINARY_DIR}/doc/mission-content.doxyfile"
+      ${MISSION_DOXYFILE_USER_CONTENT})
+
+  configure_file("${CFE_SOURCE_DIR}/cmake/cfe-common.doxyfile.in"
+    "${CMAKE_BINARY_DIR}/doc/cfe-common.doxyfile")
+
+  configure_file("${CFE_SOURCE_DIR}/cmake/osal-common.doxyfile.in"
+    "${CMAKE_BINARY_DIR}/doc/osal-common.doxyfile")
+
+  configure_file("${CFE_SOURCE_DIR}/cmake/mission-detaildesign.doxyfile.in"
+    "${CMAKE_BINARY_DIR}/doc/mission-detaildesign.doxyfile")
+
+  # Generate an "empty" osconfig.h file for doxygen purposes
+  # this does not have the actual user-defined values, but will
+  # have the documentation associated with each macro definition.
+  configure_file("${osal_MISSION_DIR}/osconfig.h.in"
+    "${CMAKE_BINARY_DIR}/doc/osconfig-example.h")
+
+  # The user guide should include the doxygen from the _public_ API files from CFE + OSAL
+  # NOTE: the userguide is built against the headers of the default core apps. Even if
+  # an alternate version of the module is in use, it should adhere to the same interface.
+  set(SUBMODULE_HEADER_PATHS
+    "${osal_MISSION_DIR}/src/os/inc/*.h"
+    "${psp_MISSION_DIR}/psp/fsw/inc/*.h"
+  )
+  foreach(MODULE core_api es evs fs msg sb tbl time)
```

Comment:
Missing SBR and resourceID modules?

Reply:
SBR and resourceID are internal to cFE, no exposed API.

Reply:
Also note that this is changed in PR #1290, to use `${MISSION_CORE_MODULES}` instead.  See https://github.com/nasa/cFE/blob/1a72fad33f9329ca66b67688d19f2c351d56405a/cmake/mission_build.cmake#L291